ark/client.py

```diff
@@ -0,0 +1,47 @@
+import inspect
+import pkgutil
+from importlib import import_module
+from pathlib import Path
+
+from ark.api.resource import Resource
+from ark.connection import Connection
+
+
+VERSION_TO_STRING_MAPPING = {
+    'v1': 'one',
+    'v2': 'two',
+}
+
+
+class ArkClient(object):
+
+    def __init__(self, host, port, nethash=None, version=None, api_version='v2'):
+        if api_version not in ['v1', 'v2']:
+            raise Exception('Only versions "v1" and "v2" are supported')
+
+        self.api_version = api_version
+
+        self.connection = Connection(self, host, port, nethash, version)
+        self.import_api()
+        self.connection.autoconfigure()
+
+    def import_api(self):
+        """
+        Dinamically imports endpoints for correct api version.
+        """
+        version = VERSION_TO_STRING_MAPPING[self.api_version]
+        # Get all modules under the wanted version folder
+        modules = pkgutil.iter_modules([Path(__file__).parent / 'api' / version])
+        for _, name, _ in modules:
+            module = import_module('.{}'.format(name), package='ark.api.{}'.format(version))
+
+            for attr in dir(module):
```

Comment:
Instead of looping through, could you use `hasattr`? Eg. `if hasattr(module, 'Resource'):`

Reply:
No, because `Resource` is a subclass of the actual class I need.